### PR TITLE
More proposed backports for 3.2.5

### PR DESCRIPTION
**Pull request by vadz**

### CODE REVIEW

**Comment by Kumazuma** [comment]
Could you add the missing default ctor of wxFileDilalog in wxMSW?

**Comment by vadz** [comment]
> Could you add the missing default ctor of wxFileDilalog in wxMSW?

Done. And while looking at this I've realized that #24113 was actually wrong and resulted in crash if you just did

```cpp
{
   wxFileDialog fd;
}
```

so I've fixed it too.

**Comment by vadz** [comment]
Oops, and somehow the tests crash too now. I can't debug this now, so I'll probably revert this after all.

File: src/msw/filedlg.cpp [hunk 1]
```diff
@@ -1028,6 +1028,11 @@ wxFileDialog::wxFileDialog(wxWindow *parent,
 
 {
     // NB: all style checks are done by wxFileDialogBase::Create
+}
+
```
Comment (Kumazuma) [inline src/msw/filedlg.cpp:1032]:
```suggestion
    m_data = NULL;
}

```
It is not initialized as NULL.

Comment (vadz) [reply src/msw/filedlg.cpp:1034]: Oh, thanks for noticing this, cherry pick removed this accidentally as it doesn't exist in master. Fixed.

**Pull request merged**